Index: domain/v1/anki.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/domain/v1/anki.go b/domain/v1/anki.go
new file mode 100644
--- /dev/null	(date 1694904546263)
+++ b/domain/v1/anki.go	(date 1694904546263)
@@ -0,0 +1,28 @@
+package domain
+
+import (
+	"anki-support/infrastructure/anki"
+	"anki-support/infrastructure/gcp"
+	"anki-support/infrastructure/openai"
+)
+
+type Anki struct {
+	anki   anki.Ankier
+	openai openai.OpenAIer
+	gcp    gcp.GCPer
+}
+
+func NewAnki(anki anki.Ankier, openai openai.OpenAIer, gcp gcp.GCPer) Ankier {
+	return &Anki{anki: anki, openai: openai, gcp: gcp}
+}
+
+type Ankier interface {
+	GetNoteListByDeckName(deckName string) (output []AnkiNote, err error)
+}
+
+func (a *Anki) GetNoteListByDeckName(s string) (output []AnkiNote, err error) {
+	return
+}
+
+type AnkiNote struct {
+}
